### PR TITLE
[native bridge smart contract 5/n] - new message types

### PR DESCRIPTION
**Pull request by patrickkuo**
## Description 

Added BlockList, UpdateBridgeLimit and UpdateAssetPrice messages

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-framework/packages/bridge/sources/bridge.move [hunk 1]
```diff
@@ -20,7 +20,7 @@ module bridge::bridge {
     use bridge::committee::{Self, BridgeCommittee};
     use bridge::message::{Self, BridgeMessage, BridgeMessageKey};
     use bridge::message_types;
-    use bridge::treasury::{Self, BridgeTreasury, token_id};
+    use bridge::treasury::{Self, BridgeTreasury};
```
Comment (patrickkuo) [inline crates/sui-framework/packages/bridge/sources/bridge.move:23]: Changes in this file are some minor clean up unrelated to the new messages changes...

File: crates/sui-framework/packages/bridge/sources/message_types.move [hunk 2]
```diff
@@ -6,6 +6,8 @@ module bridge::message_types {
     const TOKEN: u8 = 0;
     const COMMITTEE_BLOCKLIST: u8 = 1;
     const EMERGENCY_OP: u8 = 2;
+    const UPDATE_BRIDGE_LIMIT: u8 = 3;
+    const UPDATE_ASSET_PRICE: u8 = 4;
     //const COMMITTEE_CHANGE: u8 = 2;
     //const NFT: u8 = 4;
```
Comment (longbowlu) [inline crates/sui-framework/packages/bridge/sources/message_types.move:12]: remove?

File: crates/sui-framework/packages/bridge/sources/message.move [hunk 3]
```diff
@@ -83,6 +105,42 @@ module bridge::message {
...
+        // blocklist payload should consist of one byte blocklist type, and list of 33 bytes ecdsa pub keys
+        let payload_length = vector::length(&message.payload);
+        assert!(payload_length == COMPRESSED_ECDSA_PUB_KEY_LENGTH + 1, ETrailingBytes);
+        let bcs = bcs::new(message.payload);
+        let blocklist_type = bcs::peel_u8(&mut bcs);
+        let validator_pub_key = bcs::into_remainder_bytes(bcs);
+        Blocklist {
+            blocklist_type,
+            validator_pub_key
+        }
```
Comment (longbowlu) [inline crates/sui-framework/packages/bridge/sources/message.move:118]: does this take care of length of the pub keys?

Comment (longbowlu) [reply crates/sui-framework/packages/bridge/sources/message.move:118]: according to your test it looks like it does

Comment (patrickkuo) [reply crates/sui-framework/packages/bridge/sources/message.move]: not magic but bug 🤦🏻‍♂️, the test wasn't testing the extract path.....

**Review by longbowlu — APPROVED** [review]
thanks! please add more tests so we are more confident that move/solidity/rust is in sync

**Pull request merged**